src/libsyncengine/requests/serverrequests.cpp

```diff
@@ -985,6 +986,120 @@ ExitCode ServerRequests::getUserFromSyncDbId(int syncDbId, User &user) {
     return ExitCodeOk;
 }
 
+ExitCode ServerRequests::sendLogToSupport(bool includeArchivedLog, std::function<bool(char, int)> progressCallback,
+                                          ExitCause &exitCause) {
+    exitCause = ExitCauseUnknown;
+    ExitCode exitCode = ExitCodeOk;
+    std::function<bool(char, int)> safeProgressCallback =
+        progressCallback != nullptr ? std::function<bool(char, int)>([progressCallback](char status, int percent) { return progressCallback(status, percent); })
+                                    : std::function<bool(char, int)>([](char, int) { return true; });
+    
+    safeProgressCallback('A', 0);
+
+    if (bool found = false; !ParmsDb::instance()->updateAppState(AppStateKey::LastLogUploadArchivePath, "", found) || !found) {
+        LOG_WARN(Log::instance()->getLogger(), "Error in ParmsDb::updateAppState");
+        // Do not return here because it is not a critical error, especially in this context where we are trying to send logs
+    }
+
+    const SyncPath logUploadTempFolder = CommonUtility::getAppSupportDir() / "logUploadTemp";
```

Comment:
I would prefer to use IoHelper::tempDirectoryPath()

Reply:
Done in commit: [ec417fd](https://github.com/Infomaniak/desktop-kDrive/pull/117/commits/ec417fd7ce25a77d7c35f0c1096295b5259d746f)